later/lib/database/repository/record_repository.dart

```diff
@@ -1,37 +1,71 @@
-import 'dart:async';
-
 import 'package:injectable/injectable.dart';
-import 'package:isar/isar.dart';
-import 'package:later/database/models/record_model.dart';
+import 'package:later/database/datasources/records_data_source.dart';
+import 'package:later/domain/models/record_model.dart';
+import 'package:later/domain/repository/record_repository.dart';
 
 @lazySingleton
-class RecordRepository {
-  late final Isar _isar;
+class RecordRepositoryImpl implements RecordsRepository {
+  RecordRepositoryImpl({
+    required this.recordsDataSurce,
+  });
 
-  // We must preinitialize it first
-  @FactoryMethod(preResolve: true)
-  static Future<RecordRepository> init() async {
-    final repo = RecordRepository()
-      .._isar = await Isar.open([RecordModelSchema]);
+  final RecordsDataSource recordsDataSurce;
 
-    return repo;
+  @override
+  Future<void> create({
+    required String? url,
+    required String title,
+    String? description,
+    DateTime? lastEditedAt,
+  }) async {
+    try {
+      if (title.isNotEmpty) {
+        await recordsDataSurce.create(
+          url: url,
+          title: title,
+          description: description,
+          lastEditedAt: lastEditedAt,
+        );
+      }
+    } catch (error) {
+      throw Exception();
+    }
```

Comment:
The way this `try-catch` is implemented is incorrect. With this, we will lose the actual problem if one occurs. We will only get an `Exception`, not something specific. So we won't able to handle it properly on the UI layer: for example, to show a corresponding error message (like this ID is taken, or URL must be provided — just an example).

Prefer having try-catches only:
- If you handling it in a bloc. Then we **must** have a `try-catch` to ensure our app won't crash if something goes wrong, but it will show an error message or sth like this.
- If you need to remap an exception to some other exception. For example, here we should map a _local data source_ exception from the `data` layer to a _repository_ exception, still from the `data` layer. Then, we will map it to a custom `domain` layer exception. 
  
But we'll focus on the latter later. For now, remove the try-catch